adagrams/game.py

```diff
@@ -1,11 +1,163 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+# =========== WAVE ONE ===========
+# ================================
+
 def draw_letters():
-    pass
+    
+    letters = []
+    
+    count = 0
+    
+    new_letter_dict = LETTER_POOL.copy()
+
+    while len(letters) < 10:
+        random_letter = random.choice(list(new_letter_dict.keys()))
+        for letter, points in new_letter_dict.items():
+            if letter == random_letter and points > 1:
+                letters.append(random_letter)
+                points -= 1
+                new_letter_dict[letter] = points
+            else:
+                continue
+        count += 1 
+    
+    return letters
+
+# =========== WAVE TWO ===========
+# ================================
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+
+    list_of_words = list(word)
+
+    frequency_in_word = {}
+
+    for letter in list_of_words:
+        if letter in frequency_in_word:
+            frequency_in_word[letter] += 1
+        else:
+            frequency_in_word[letter] = 1
+
+    frequency_in_bank = {}
+
+    for letter in letter_bank:
+        if letter in frequency_in_bank:
+            frequency_in_bank[letter] += 1
+        else:
+            frequency_in_bank[letter] = 1
+
+
+    common = frequency_in_word.items() & frequency_in_bank.items() 
+
+    if len(common) == len(frequency_in_word):
+        return True
+    else:
+        return False
```

Comment:
This is an interesting implementation that mostly works, but there's a few style things I want to point out, as well as a logic error that sadly our tests did not account for.